### PR TITLE
Feat/video link added to online meeting

### PR DESCRIPTION
**Pull request by adnanalicic**
Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: src/main/java/de/caritas/cob/userservice/api/adapters/web/mapping/AppointmentDtoMapper.java [hunk 1]
```diff
@@ -22,7 +31,22 @@ public Map<String, Object> mapOf(Appointment appointment, AuthenticatedUser auth
...
+    if (authenticatedUser.getRoles().contains(UserRole.CONSULTANT.getValue())
+        && appointment.getConsultantEmail() == null) {
+      appointmentMap.put("consultantId", authenticatedUser.getUserId());
+    } else if (authenticatedUser.getRoles().contains(UserRole.TECHNICAL.getValue())
+        && appointment.getConsultantEmail() != null) {
+      Optional<Consultant> consultant =
+          consultantRepository.findByEmailAndDeleteDateIsNull(appointment.getConsultantEmail());
+      if (!consultant.isPresent()) {
+        throw new BadRequestException(
+            "Consultant doesn't exist for give email " + appointment.getConsultantEmail());
```
Comment (tkuzynow) [inline src/main/java/de/caritas/cob/userservice/api/adapters/web/mapping/AppointmentDtoMapper.java]: typo, should be given

File: src/main/java/de/caritas/cob/userservice/api/adapters/web/mapping/AppointmentDtoMapper.java [hunk 2]
```diff
@@ -22,7 +31,22 @@ public Map<String, Object> mapOf(Appointment appointment, AuthenticatedUser auth
...
+      }
+      appointmentMap.put("consultantId", consultant.get().getId());
```
Comment (tkuzynow) [inline src/main/java/de/caritas/cob/userservice/api/adapters/web/mapping/AppointmentDtoMapper.java]: you could also use orElseThrow to achieve similar behaviour as you have above

File: src/main/java/de/caritas/cob/userservice/api/adapters/web/mapping/AppointmentDtoMapper.java [hunk 3]
```diff
@@ -4,16 +4,25 @@
...
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
 @Service
+@RequiredArgsConstructor
 public class AppointmentDtoMapper {
 
+  private final ConsultantRepository consultantRepository;
+
   public Map<String, Object> mapOf(Appointment appointment, AuthenticatedUser authenticatedUser) {
```
Comment (tkuzynow) [inline src/main/java/de/caritas/cob/userservice/api/adapters/web/mapping/AppointmentDtoMapper.java]: I think in general this method should have another parameter Optional<Long> consultantId, and the logic to find Consultant based on the authenticated user role should be placed outside of the DtoMapper, either in the calling controller or even in separate service.
DTOMappers should stay as simple as possible, as nobody expects they contain some consultant find logic.

File: src/main/java/de/caritas/cob/userservice/api/adapters/web/mapping/AppointmentDtoMapper.java (hunk 1 above)
Comment (adnanalicic) [reply src/main/java/de/caritas/cob/userservice/api/adapters/web/mapping/AppointmentDtoMapper.java]: done

File: src/main/java/de/caritas/cob/userservice/api/adapters/web/mapping/AppointmentDtoMapper.java (hunk 2 above)
Comment (adnanalicic) [reply src/main/java/de/caritas/cob/userservice/api/adapters/web/mapping/AppointmentDtoMapper.java]: ok

File: src/main/java/de/caritas/cob/userservice/api/adapters/web/mapping/AppointmentDtoMapper.java (hunk 3 above)
Comment (adnanalicic) [reply src/main/java/de/caritas/cob/userservice/api/adapters/web/mapping/AppointmentDtoMapper.java]: done

**Review by tkuzynow — APPROVED** [review]

**Pull request merged**